src/motor-operations.cc

```diff
@@ -152,6 +188,38 @@ void MotionQueueMotorOperations::EnqueueInternal(const LinearSegmentSteps &param
   new_element.state = STATE_FILLED;
   backend_->MotorEnable(true);
   backend_->Enqueue(&new_element);
+
+  // Shrink the queue
+  // TODO: We need to find a way to get the maximum number of elements
+  // of the shadow queue (ie backend_->GetQueueStats()?)
+  const unsigned int buffer_size = backend_->GetPendingElements(NULL);
+  shadow_queue_->resize(buffer_size);
+}
+
+void MotionQueueMotorOperations::GetRealtimeStatus(PhysicalStatus *status) {
+  // Shrink the queue
+
+  uint32_t loops;
+  const unsigned int buffer_size = backend_->GetPendingElements(&loops);
+  assert(buffer_size <= shadow_queue_->size());
+  if (shadow_queue_->size() > 1) shadow_queue_->resize(buffer_size);
```

Comment:
So this resize() business seems a bit hand-wavy to me (here, and in the EnqueInternal()).

This is because you want to try hard to avoid knowing the queue-size of the underlying MotionQueue so do all kinds of observational second guessing here. But it would be so much easier (and easier for someone reading the code) if you just kept track of it with a ring-buffer.  So I would suggest that either
  * you pass the size of the queue len to the constructor of the MotionQueueMotorOperations
  * or have a size_t MotionQueue::GetFixedInternalQueueSize() const; that operations downstream (such as this MotionQueueMotorOperations) can query and set-up their fixed